oauth2/src/main/java/com/linecorp/armeria/client/auth/oauth2/AbstractOAuth2AuthorizationGrant.java

```diff
@@ -108,16 +93,114 @@ private static boolean shallObtainInsteadOfUpdate(Throwable throwable) {
             @Nullable GrantedOAuth2AccessToken token);
 
     /**
-     * Produces valid OAuth 2.0 Access Token.
-     * Returns cached access token if previously obtained from the token end-point.
-     * Optionally loads access token from longer term storage provided by registered {@link Supplier}.
-     * If access token has not previously obtained, obtains is from the OAuth 2.0 token end-point using
-     * dedicated single-thread {@link ExecutorService} which makes sure all token obtain and refresh requests
-     * executed serially.
-     * Validates access token and refreshes it if necessary.
+     * Issues an {@code OAuth 2.0 Access Token} and cache it in memory and returns the cached one
+     * until the token is considered valid. It automatically refreshes the cached token once it's
+     * considered expired and returns the refreshed one.
+     *
+     * <p>Renewing a token is guaranteed to be atomic even though the method is invoked by multiple threads.
+     *
+     * <p>It optionally tries to load an access token from {@link #fallbackTokenProvider}
```

Comment:
Javadoc cannot link to non-public methods or fields.
```suggestion
     * <p>It optionally tries to load an access token from
     * {@link OAuth2ClientCredentialsGrantBuilder#fallbackTokenProvider(Supplier)} or 
     * {@link OAuth2ResourceOwnerPasswordCredentialsGrantBuilder#fallbackTokenProvider(Supplier)}
```

Reply:
I didn't know that, thank you for point it out!
Updated javadoc.